lib/spack/spack/build_environment.py

```diff
@@ -716,6 +717,10 @@ def setup_package(pkg, dirty):
                 load_module("cce")
             load_module(mod)
 
+        if os.environ.get("CRAY_CPU_TARGET") == "mic-knl":
+            if "cce" not in pkg.compiler.modules:
+                unload_module("cce")
```

Comment:
Should the `cce` module not be loaded at all unless it is in `pkg.compiler.modules`?

Reply:
@scheibelp cce is Cray's own compilers and they carry with them a set of binutils. It's loaded above the earlier addition purely to get around a one-line bug in Cray's cray-mic-knl module. Unless users are loading it as a compiler, there's no legitimate reason for it to be in the build environment.